### DISCUSSION TITLE
Upgrad from 1.4.9 to 3.13.10

### DISCUSSION THREAD

**Discussion by wrmilliron** [Kong/unirest-java#447] [Q&A]
I inherited a project that used 1.4.9 to make REST calls. As part of an effort to enable TLS, unfortunatelly with a self signed cert, i believe there is now a config I can set to help ingonre the private cert issues. I do have the cert in my java cacertsstor  and I have things partially working with 1.4.9 but subsequent REST calls give me a  this:

PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

Wierd thing is it looks like the very first REST call works.

Anyways I figure probably bbet to get a more current version of unirest. I updated by build.gradle file and it seems to be pulling down a new version. Plus I saw something about a config to maybe ignore SSL verification?

The issue is all the import stateemnts for unitest are no longer valid.I dod look at the osurce and thought I had the right imports but no joy.
For example import com.mashape.unirest.HttpResponse;

I tried a number of variations based on the unirest source but no joy.

Looking at the most recent documentation it looks the the usage has not changed.

Stuck with java 1.8 unfortunatelly.

Is there a code complete Helloworld out there with all the import statements?
Do I need to add more to my build.gradle other than changing the dependancy to implementation 'com.konghq:unirest-java:3.13.10' ?

Anyways I figured I would ask before exploring other options. Maybe its a simple change :)

**Comment by ryber** [comment]
there is an upgrade guide here: https://github.com/Kong/unirest-java/blob/main/UPGRADE_GUIDE.md

the main package is ```kong.unirest``` if you are using Intellij or another good IDE it will find them and add the imports for you.

For the certs you can either disable (sad) or load the self signed cert into the cert store. Both examples are in this test:

https://github.com/Kong/unirest-java/blob/b8cc0b0d66c014a1206cdcf4c00f1ac049685e5b/unirest/src/test/java/BehaviorTests/CertificateTests.java

**Reply by wrmilliron** [reply]
I saw the upgrade guide and I must have not grok'd something and I will revisit it.
For some reason intellij was not finding the right import stuff even when I blew away my gradle cache.
In my gradle cache I see unirest-java and unirest-java-parent. Hoping that is what I should see.
I will dig into the tests. 
Thanks for the reply. I am hoping it is something really simple :)

**Reply by wrmilliron** [reply]
but I should be able to do something like import kong.unirest.HttpResponse;  ?
I am guessing something is just munged in my intellij.

**Reply by wrmilliron** [reply]
I did get it working so far for the non-SSL connection with unirest 3.10.x. Lost of phun getting Gradle and Intellij to both agree to use the new version then converting everything to use the unirest JSON stuff. The latter was just mildly tedious.
Dealng with SSL is a Monday issue.
